source/server/config/network/original_dst.cc

```diff
@@ -0,0 +1,47 @@
+#include <string>
+
+#include "envoy/registry/registry.h"
+#include "envoy/server/filter_config.h"
+
+#include "common/config/well_known_names.h"
+#include "common/filter/original_dst.h"
+
+namespace Envoy {
+namespace Server {
+namespace Configuration {
+
+/**
+ * Config registration for the original dst filter. @see NamedNetworkFilterConfigFactory.
+ */
+class OriginalDstConfigFactory : public NamedListenerFilterConfigFactory {
+public:
+  // NamedListenerFilterConfigFactory
+  ListenerFilterFactoryCb createFilterFactory(const Json::Object&, FactoryContext&) override {
+    return [](Network::ListenerFilterManager& filter_manager) -> void {
+      filter_manager.addAcceptFilter(Network::ListenerFilterSharedPtr{new Filter::OriginalDst()});
+    };
+  }
+
+  ListenerFilterFactoryCb createFilterFactoryFromProto(const Protobuf::Message&,
+                                                       FactoryContext&) override {
+    return [](Network::ListenerFilterManager& filter_manager) -> void {
+      filter_manager.addAcceptFilter(Network::ListenerFilterSharedPtr{new Filter::OriginalDst()});
+    };
+  }
+
+  ProtobufTypes::MessagePtr createEmptyConfigProto() override {
+    return ProtobufTypes::MessagePtr{new Envoy::ProtobufWkt::Empty()};
```

Comment:
nit: `std::make_unique`

Reply:
Envoy::ProtobufWkt::Empty() is a more specific type than ProtobufTypes::Message, do you think using std::make_unique would make this easier on the eyes?